service/src/test/java/org/kiwiproject/champagne/dao/HostDaoTest.java

```diff
@@ -153,4 +151,172 @@ void shouldReturnEmptyWhenNoHostFound() {
         }
     }
 
+    @Nested
+    class UpdateTagList {
+        @Test
+        void shouldAddGivenTagsWhenNoneExist() {
+            var systemId = insertDeployableSystem(handle, "kiwi");
+            var envId = insertDeploymentEnvironmentRecord(handle, "dev", systemId);
+            var hostId = insertHostRecord(handle, "localhost", envId, systemId);
+            var tagId = insertTagRecord(handle, "core", systemId);
+
+            dao.updateTagList(hostId, List.of(tagId));
+
+            var tag_ids = handle.select("select tag_id from host_tags where host_id = ?", hostId)
+                    .mapTo(Long.class)
+                    .list();
+
+            assertThat(tag_ids).contains(tagId);
+        }
+
+        @Test
+        void shouldRemoveAllTagsIfNoneGiven() {
+            var systemId = insertDeployableSystem(handle, "kiwi");
+            var envId = insertDeploymentEnvironmentRecord(handle, "dev", systemId);
+            var hostId = insertHostRecord(handle, "localhost", envId, systemId);
+            var tagId = insertTagRecord(handle, "core", systemId);
+
+            handle.createUpdate("insert into host_tags "
```

Comment:
This and similar multi-line strings can be changed to a _text block_ (assuming I am correct that this service targets JDK 17). So:

```java
"""insert into host_tags
    (host_id, tag_id)
    values
   (:hostId, :tagId)""" 
```

I am a bit surprised IntelliJ didn't suggest it. Maybe there is some intention action that is not enabled by default.